src/executables/master_heartbeat.c

```diff
@@ -6598,6 +6605,88 @@ hb_help_sprint_jobs_info (HB_JOB * jobs, char *buffer, int max_length)
   return p - buffer;
 }
 
+/**
+ * Checks if a host name is canonical name (if contains a '.' (dot))
+ *
+ * @param hostname host name to check
+ * @param pos (out) first occurrence of the '.' (dot) character
+ *
+ * @return true if host name is canonical name, false otherwise
+ * NOTE: if return is true then also pos output param is set to the first occurrence of the '.' (dot) character
+ */
+static bool
+is_canonical_name (const char *hostname, size_t * pos)
+{
+  *pos = 0;
+  size_t str_len = strlen (hostname);
+  for (size_t i = 0; i < str_len; ++i)
+    {
+      if (hostname[i] == '.')
+	{
+	  *pos = i;
+	  return true;
+	}
+    }
+
+  return false;
+}
+
+/**
+ * Compare two host names if are equal, if one of the host names is canonical name and the other is not, then
+ * only host part (e.g. for canonical name "host-1.cubrid.org" host part is "host-1") is used for comparison
+ *
+ * for example following hosts are equal:
+ *  "host-1"            "host-1"
+ *  "host-1"            "host-1.cubrid.org"
+ *  "host-1.cubrid.org" "host-1"
+ *  "host-1.cubrid.org" "host-1.cubrid.org"
+ *
+ * for example following hosts are not equal:
+ *  "host-1"            "host-2"
+ *  "host-1.cubrid.org" "host-2"
+ *  "host-1"            "host-2.cubrid.org"
+ *  "host-1.cubrid.org" "host-2.cubrid.org"
+ *  "host-1.cubrid.org" "host-1.cubrid.com"
+ *
+ * @param hostname_a first hostname
+ * @param hostname_b second hostname
+ *
+ * @return true if hostname_a is same as hostname_b
+ */
+static bool
+are_hostnames_equal (const char *hostname_a, const char *hostname_b)
+{
+  size_t dot_pos_a = 0;
+  bool is_canonical_name_a = is_canonical_name (hostname_a, &dot_pos_a);
+
+  size_t dot_pos_b = 0;
+  bool is_canonical_name_b = is_canonical_name (hostname_b, &dot_pos_b);
+
+  if (is_canonical_name_a && !is_canonical_name_b)
+    {
+      // first is canonical and second is not
+      char tmp_hostname_a[MAXHOSTNAMELEN];
```

Comment:
Could `<cstring>/std::strncmp` be used?

Reply:
I think is not the case, at the end I used `strcmp` not `strncmp`